src/briefcase/integrations/subprocess.py

```diff
@@ -53,61 +53,108 @@ def run(self, args, **kwargs):
+            self._log_return_code(exc.returncode)
+            raise
+
+        self._log_return_code(command_result.returncode)
+        return command_result
 
     def check_output(self, args, **kwargs):
-        """A wrapper for subprocess.check_output()
-
-        If verbosity >= 2, the executed command will be printed to the console.
+        """
+        A wrapper for subprocess.check_output()
 
         The behavior of this method is identical to subprocess.check_output(),
         except for the `env` argument. If provided, the current system
         environment will be copied, and the contents of env overwritten
         into that environment.
         """
-        # Invoke subprocess.check_output
-        if self.command.verbosity >= 2:
-            print(">>> {cmdline}".format(
-                cmdline=' '.join(shlex.quote(arg) for arg in args)
-            ))
+        subprocess_kwargs = self.final_kwargs(**kwargs)
+        self._log_command(args)
+        self._log_environment(subprocess_kwargs)
 
-        return self._subprocess.check_output(
-            [
-                str(arg) for arg in args
-            ],
-            **self.final_kwargs(**kwargs)
-        )
+        try:
+            cmd_output = self._subprocess.check_output(
+                [
+                    str(arg) for arg in args
+                ],
+                **subprocess_kwargs
+            )
+        except subprocess.CalledProcessError as exc:
+            self._log_output(exc.output, exc.stderr)
+            self._log_return_code(exc.returncode)
+            raise
+
+        self._log_output(cmd_output)
+        self._log_return_code(0)
+        return cmd_output
 
     def Popen(self, args, **kwargs):
-        """A wrapper for subprocess.Popen()
-
-        If verbosity >= 2, the executed command will be printed to the console.
+        """
+        A wrapper for subprocess.Popen()
 
         The behavior of this method is identical to subprocess.Popen(),
         except for the `env` argument. If provided, the current system
         environment will be copied, and the contents of env overwritten
         into that environment.
         """
-        # Invoke subprocess.check_output
-        if self.command.verbosity >= 2:
-            print(">>> {cmdline}".format(
-                cmdline=' '.join(shlex.quote(arg) for arg in args)
-            ))
+        subprocess_kwargs = self.final_kwargs(**kwargs)
+        self._log_command(args)
+        self._log_environment(subprocess_kwargs)
 
         return self._subprocess.Popen(
             [
                 str(arg) for arg in args
             ],
-            **self.final_kwargs(**kwargs)
+            **subprocess_kwargs
         )
+
+    def _log_command(self, args):
+        """
+        Log the entire console command being executed.
+        """
+        cmdline = ' '.join(shlex.quote(str(arg)) for arg in args)
+        self.command.logger.debug("\nRunning Command:")
```

Comment:
Why the `\n` here? Is this to get around the "we might have been mid-progress bar" problem?

Reply:
This is a carryover from when I was trying to decide the best way to introduce empty lines in log output; whether to make empty calls to `Log` or put line breaks directly in to the message. Given all the existing empty `print` calls, I eventually just went with empty `Log` calls.

I updated this to remove the `\n` and call `logger.debug()` instead.

The addition of this empty log output was driven by demarcating the logging of multiple shell commands running back to to back.

For instance:
```
❯ briefcase create -vv
>>> 
>>> Running Command:
>>>     docker --version
>>> Environment:
>>>     CLUTTER_IM_MODULE=xim
[OMITTED]
>>>     _=/home/user/github/briefcase/venv-3.9/bin/briefcase
>>> Command Output:
>>>     Docker version 20.10.14, build a224086
>>> Return code: 0
>>> 
>>> Running Command:
>>>     docker info
>>> Environment:
>>>     CLUTTER_IM_MODULE=xim
[OMITTED]
>>>     _=/home/user/github/briefcase/venv-3.9/bin/briefcase
>>> Command Output:
>>>     Client:
>>>      Context:    default
>>>      Debug Mode: false
>>>      Plugins:
>>>       app: Docker App (Docker Inc., v0.9.1-beta3)
>>>       buildx: Docker Buildx (Docker Inc., v0.8.1-docker)
>>>       scan: Docker Scan (Docker Inc., v0.17.0)
>>>     
>>>     Server:
>>>      Containers: 106
>>>       Running: 0
>>>       Paused: 0
>>>       Stopped: 106
>>>      Images: 71
>>>      Server Version: 20.10.14
>>>      Storage Driver: overlay2
>>>       Backing Filesystem: extfs
>>>       Supports d_type: true
>>>       Native Overlay Diff: true
>>>       userxattr: false
>>>      Logging Driver: json-file
>>>      Cgroup Driver: cgroupfs
>>>      Cgroup Version: 1
>>>      Plugins:
>>>       Volume: local
>>>       Network: bridge host ipvlan macvlan null overlay
>>>       Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
>>>      Swarm: inactive
>>>      Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
>>>      Default Runtime: runc
>>>      Init Binary: docker-init
>>>      containerd version: 3df54a852345ae127d1fa3092b95168e4a88e2f8
>>>      runc version: v1.0.3-0-gf46b6ba
>>>      init version: de40ad0
>>>      Security Options:
>>>       apparmor
>>>       seccomp
>>>        Profile: default
>>>      Kernel Version: 5.4.0-109-generic
>>>      Operating System: Ubuntu 18.04.6 LTS
>>>      OSType: linux
>>>      Architecture: x86_64
>>>      CPUs: 4
>>>      Total Memory: 9.728GiB
>>>      Name: virt-ubuntu18
>>>      ID: 2O2U:B6J2:M55X:PQ6G:SGSF:3RDN:H2GE:XV37:3F6N:GRPV:JGY2:Y2D6
>>>      Docker Root Dir: /var/lib/docker
>>>      Debug Mode: false
>>>      Registry: https://index.docker.io/v1/
>>>      Labels:
>>>      Experimental: false
>>>      Insecure Registries:
>>>       127.0.0.0/8
>>>      Live Restore Enabled: false
>>>     
>>>     WARNING: No swap limit support
>>> Return code: 0

Application helloworld already exists; overwrite [y/N]? 
```

Current output for posterity:
```
❯ briefcase create -vv
>>> docker --version
>>> docker info

Application helloworld already exists; overwrite [y/N]?
```

Reply:
Ok - that makes sense. I'll probably tweak this when I land so that a "clearing debug newline" (i.e., debug with *no* input) omits the `>>>` preface; this has the added benefit that if you call multiple commands in a row, there's even better demarcation between calls.